openlibrary/core/bookshelves_events.py

```diff
@@ -124,6 +124,22 @@ def select_distinct_by_user_type_and_year(cls, username, event_type, year):
 
         return list(oldb.query(query, vars=data))
 
+    @classmethod
+    def total_yearly_reading_goals(cls, since: date | None = None) -> int:
+        """Returns (int) number of reading goals updated. `since` may be used
+        to limit the result to those reading goals updated since a specific
+        date. Any python datetime.date type should work.
+
+        :param since: returns all reading goals after date
+        """
+        oldb = db.get_db()
+
+        query = "SELECT count(*) from yearly_reading_goals"
+        if since:
+            query += " WHERE updated >= $since"
+        results = tuple(oldb.query(query, vars={'since': since}))
+        return int(results[0]['count(*)'])
```

Comment:
```suggestion
        return int(results[0])
```
I can see you were taking inspiration from [`total_books_logged()`](https://github.com/internetarchive/openlibrary/blob/51f29488b3bd9adfc502d5c4e1f3d8fc25ef4c9c/openlibrary/core/bookshelves.py#L51-L79), but I think that function may be misleading us a bit. 

Curiously, this highlights exactly the issue @cclauss raised. In spite of `total_books_logged()` claiming it returns an `int`, I think it returns `Storage` (which is a custom type that is close to a `dict`).

I can see you may have detected this, and you were on the right track of you wanted an `int` out of it. Instead of `results[0]['count(*)']`, you'd want `results[0]['count']`. Except here you actually do want the storage object, I think.

The reasoning for that is a bit attenuated, but recall that `storage` is like a dictionary, and eventually this code ends up passed to [`openlibrary/templates/admin/index.html`](https://github.com/internetarchive/openlibrary/blob/51f29488b3bd9adfc502d5c4e1f3d8fc25ef4c9c/openlibrary/templates/admin/index.html#L159-L165) and you can see there our `total_books_logged` example has `<td class="amount">$commify(counts.reading_log['total_books_logged']['week']['count'])</td>`.